pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -176,7 +179,12 @@ def advance(self, data_fetcher: AbstractDataFetcher) -> None:  # type: ignore[ov
             )
 
             # hook
-            response = self.trainer.call_hook("on_train_batch_start", batch, batch_idx, **extra_kwargs)
+            self.trainer._call_callback_hooks("on_train_batch_start", batch, batch_idx, **extra_kwargs)
+            model_response = self.trainer._call_lightning_module_hook(
+                "on_train_batch_start", batch, batch_idx, **extra_kwargs
+            )
+            ttp_response = self.trainer._call_ttp_hook("on_train_batch_start", batch, batch_idx, **extra_kwargs)
+            response = ttp_response if model_response is None else model_response
```

Comment:
The TTP is annotated as returning `None`

```suggestion
            response = self.trainer._call_lightning_module_hook(
                "on_train_batch_start", batch, batch_idx, **extra_kwargs
            )
            self.trainer._call_ttp_hook("on_train_batch_start", batch, batch_idx, **extra_kwargs)            
```

Reply:
it's not directly annotated as returning None, it is just missing the return type annotation. The reason it is missing the return type annotation is because it previously had `-> Optional[Any]` as return type but this caused a mypy error:
`pytorch_lightning/loops/optimization/optimizer_loop.py:439: error: Argument 1 to "_extract_hiddens" has incompatible type "Optional[Any]"; expected "Union[Tensor, Dict[str, Any]]"  [arg-type]`

so I removed it. But in order to maintain the current behavior of call_hook, ttp hooks _should_ be allowed to return something.